cmd/process-agent/collector_test.go

```diff
@@ -111,32 +113,29 @@ func TestDisableRealTime(t *testing.T) {
 		{
 			name:            "true",
 			disableRealtime: true,
-			expectedChecks:  []checks.Check{checks.Process, checks.Container},
+			expectedChecks:  []checks.Check{checks.Container},
 		},
 		{
 			name:            "false",
 			disableRealtime: false,
-			expectedChecks:  []checks.Check{checks.Process, checks.Container, checks.RTContainer},
+			expectedChecks:  []checks.Check{checks.Container, checks.RTContainer},
 		},
 	}
 
 	assert := assert.New(t)
 	cfg := config.NewDefaultAgentConfig()
-	cfg.EnabledChecks = []string{
-		config.ProcessCheckName,
-		config.RTProcessCheckName,
-		config.ContainerCheckName,
-		config.RTContainerCheckName,
-	}
 
 	for _, tc := range tests {
 		t.Run(tc.name, func(t *testing.T) {
 			mockConfig := ddconfig.Mock()
 			mockConfig.Set("process_config.disable_realtime_checks", tc.disableRealtime)
+			mockConfig.Set("process_config.process_discovery.enabled", false) // Not an RT check so we don't care
+
+			enabledChecks := getChecks(&sysconfig.Config{}, &oconfig.OrchestratorConfig{}, true)
+			assert.EqualValues(tc.expectedChecks, enabledChecks)
 
-			c, err := NewCollector(cfg)
+			c, err := NewCollector(cfg, enabledChecks)
 			assert.NoError(err)
-			assert.ElementsMatch(tc.expectedChecks, c.enabledChecks)
```

Comment:
Can we keep this check since `Collector` still has a `enabledChecks` field?

Reply:
`Collector.enabledChecks` is really just a pointer to `enabledChecks`, so it wouldn't be testing much, however I don't think it could hurt!